src/main/java/com/iexec/resultproxy/ipfs/IpfsService.java

```diff
@@ -62,4 +60,29 @@ public static boolean isIpfsHash(String hash) {
         }
     }
 
+    @Override
+    @Retryable(maxAttempts = 10,
+            backoff = @Backoff(
+                    delay = 1000
+            ))
+    public void start() {
+        this.ipfs = new IPFS(multiAddress);
+    }
+
+    @Recover
+    public void start(Throwable t) {
+        log.error("Exception when initializing IPFS [exception:{}]", t.getMessage());
+        log.warn("Shutting down the service since IPFS is necessary");
+        System.exit(1);
+    }
```

Comment:
I find the `System.exit` dangerous here and would prefer an Exception to have a full stack trace and its cause
```suggestion
    @Recover
    public void start(Throwable t) {
        log.error("Exception when initializing IPFS", t);
        throw new IllegalStateException("Exception when initializing IPFS", t);
    }
```
or if the logs are not mandatory, we could remove this entirely and let the `RuntimeException` be fired by `new IPFS(multiAddress)` after the 10th failed try.

Reply:
I'd prefer to keep the logs, so let's re-throw the exception.